plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1554,6 +1554,24 @@ public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHan
         }
     }
 
+    @Override
+    public void setColumnType(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle columnHandle, io.trino.spi.type.Type type)
+    {
+        IcebergTableHandle table = (IcebergTableHandle) tableHandle;
+        IcebergColumnHandle column = (IcebergColumnHandle) columnHandle;
+        verify(column.isBaseColumn(), "Cannot change nested field types");
+
+        Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
+        try {
+            icebergTable.updateSchema()
+                    .updateColumn(column.getName(), toIcebergType(type).asPrimitiveType())
+                    .commit();
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to set column type: " + firstNonNull(e.getMessage(), e), e);
```

Comment:
```
create table test1(col1 int, col2 varchar) with (partitioning=array['col1']);
alter table test1 alter column col1 set data type bigint;

alter table test1 alter column col1 set data type varchar;
Query 20230116_064352_00012_rjq4n failed: Failed to set column type: Cannot change column type: col1: long -> string
io.trino.spi.TrinoException: Failed to set column type: Cannot change column type: col1: long -> string
	at io.trino.plugin.iceberg.IcebergMetadata.setColumnType(IcebergMetadata.java:1571)
```

`Cannot change column type: col1: long -> string` - not the best error message , but still better than nothing.
No change requested.